tests/pages/firefox/partners.py

```diff
@@ -0,0 +1,168 @@
+            item.click()
+            self.wait.until(lambda s: self.page.is_phone_os_screen_displayed)
+
+        def show_marketplace(self):
+            el = self.find_element(self._marketplace_locator)
+            item = self.MenuItem(self.page, root=el)
+            item.click()
+            self.wait.until(
+                lambda s: self.page.is_phone_marketplace_screen_displayed)
+
+        def show_android(self):
+            el = self.find_element(self._android_locator)
+            item = self.MenuItem(self.page, root=el)
+            item.click()
+            self.wait.until(lambda s: self.page.is_android_phone_displayed)
+
+        class MenuItem(FirefoxBasePageRegion):
+
+            _link_locator = (By.TAG_NAME, 'a')
+
+            @property
+            def is_selected(self):
+                return 'active' in self._root.get_attribute('class')
+
+            def click(self):
+                self.find_element(self._link_locator).click()
+                self.wait.until(lambda s: self.is_selected)
+
+    class MWCMenu(FirefoxBasePageRegion):
+
+        _root_locator = (By.ID, 'mwc-menu')
+        _map_locator = (By.ID, 'menu-mwc-map')
+        _schedule_locator = (By.ID, 'menu-mwc-schedule')
+
+        def show_map(self):
+            modal = Modal(self)
+            el = self.find_element(self._map_locator)
+            self.MenuItem(self.page, root=el).click()
+            self.wait.until(lambda s: modal.is_displayed)
+            return modal
+
+        def show_schedule(self):
+            modal = Modal(self)
+            el = self.find_element(self._schedule_locator)
+            self.MenuItem(self.page, root=el).click()
+            self.wait.until(lambda s: modal.is_displayed)
+            return modal
+
+        class MenuItem(FirefoxBasePageRegion):
+
+            _link_locator = (By.TAG_NAME, 'a')
+
+            def click(self):
+                self.find_element(self._link_locator).click()
+
+    class DevicesMenu(FirefoxBasePageRegion):
+
+        _root_locator = (By.ID, 'devices-menu')
+        _devices_locator = (By.ID, 'menu-devices')
+
+        def show_devices(self):
+            el = self.find_element(self._devices_locator)
+            self.MenuItem(self.page, root=el).click()
+            from pages.firefox.os.devices import DevicesPage
+            return DevicesPage(self.base_url, self.selenium)
+
+        class MenuItem(FirefoxBasePageRegion):
+
+            _link_locator = (By.TAG_NAME, 'a')
+
+            def click(self):
+                self.find_element(self._link_locator).click()
+
+    class ArticleMenu(FirefoxBasePageRegion):
+
+        _items_locator = (By.TAG_NAME, 'a')
+
+        @property
+        def items(self):
+            els = self.find_elements(self._items_locator)
+            return [self.MenuItem(self.page, root=el) for el in els]
+
+        class MenuItem(FirefoxBasePageRegion):
+
+            def click(self):
+                self._root.click()
```

Comment:
Ideally I'd have a wait here for the appropriate section to be displayed. Unfortunately the only way I can see to do this is to link the values of the data-section attributes of two elements. If possible, adding a class or attribute to the menu item to indicate that it is the current item (after any animations) would allow me to improve this.